serf/config.go

```diff
@@ -275,6 +284,18 @@ func DefaultConfig() *Config {
 		panic(err)
 	}
 
+	//Do we need to do this here if it is set to true in the config?
+	if len(hostname) > MaxNodeName {
+		//TODO(schristoff): panic seems a little harsh here?
+		panic(fmt.Errorf("NodeName is %v characters. "+
+			"Valid length is between 1 and 128 characters", len(hostname)))
+	}
+	var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\\-]+`)
+	if InvalidNameRe.MatchString(hostname) {
+		panic(fmt.Errorf("NodeName contains invalid characters %v , Valid characters include "+
+			"all alpha-numerics and dashes.", hostname))
+	}
```

Comment:
I think we have to do this validation in Create(), or else it won't be checked when config gets passed in from Consul. We won't have to check it here and potentially panic then either, since it'll just return an error when trying to create the serf instance.